### ISSUE TITLE
[Bug]: DATA RACE

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#13434]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 9663b5a
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

https://github.com/matrixorigin/matrixone/actions/runs/7190766771

[multi-cn-e2e-with-race-linux-x86-reports.zip](https://github.com/matrixorigin/matrixone/files/13657878/multi-cn-e2e-with-race-linux-x86-reports.zip)

```
==================
WARNING: DATA RACE
Write at 0x00c01c8c4bc8 by goroutine 216076:
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*shard[...]).Set()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/shard.go:30 +0x24a
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*LRU[...]).Set()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/lru.go:77 +0x2dd
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*LRU[...]).Set()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/lru.go:72 +0x170
  github.com/matrixorigin/matrixone/pkg/fileservice.(*MemCache).Update()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/mem_cache.go:187 +0x61b
  github.com/matrixorigin/matrixone/pkg/fileservice.(*LocalFS).Read.func3()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/local_fs.go:330 +0x131
  runtime.deferreturn()
      /usr/local/go/src/runtime/panic.go:476 +0x32
  github.com/matrixorigin/matrixone/pkg/objectio.ReadMultiBlocksWithMeta()
      /home/runner/_work/matrixone/matrixone/head/pkg/objectio/funcs.go:260 +0x433
  github.com/matrixorigin/matrixone/pkg/objectio.(*objectReaderV1).ReadMultiBlocks()
      /home/runner/_work/matrixone/matrixone/head/pkg/objectio/reader.go:302 +0x2f6
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.prefetchJob.func2()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/blockio/pipeline.go:182 +0x17e
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*Job).Run()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/job.go:122 +0x310
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*Job).Run-fm()
      <autogenerated>:1 +0x44
  github.com/panjf2000/ants/v2.(*goWorker).run.func1()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/worker.go:67 +0x1c4

Previous read at 0x00c01c8c4bc8 by goroutine 216072:
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*shard[...]).Get()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/shard.go:98 +0x24d
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*LRU[...]).Get()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/lru.go:85 +0x2ac
  github.com/matrixorigin/matrixone/pkg/fileservice/lrucache.(*LRU[...]).Get()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/lrucache/lru.go:80 +0x125
  github.com/matrixorigin/matrixone/pkg/fileservice.(*MemCache).Read()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/mem_cache.go:140 +0x7dd
  github.com/matrixorigin/matrixone/pkg/fileservice.(*LocalFS).Read()
      /home/runner/_work/matrixone/matrixone/head/pkg/fileservice/local_fs.go:323 +0xafc
  github.com/matrixorigin/matrixone/pkg/objectio.ReadOneBlockWithMeta()
      /home/runner/_work/matrixone/matrixone/head/pkg/objectio/funcs.go:186 +0xfeb
  github.com/matrixorigin/matrixone/pkg/objectio.ReadOneBlock()
      /home/runner/_work/matrixone/matrixone/head/pkg/objectio/funcs.go:119 +0x2b2
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.LoadColumnsData()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/blockio/funcs.go:45 +0x54c
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.LoadColumns()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/blockio/funcs.go:71 +0x284
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.LoadPersistedColumnDatas()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tables/utils.go:88 +0xb08
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*baseBlock).ResolvePersistedColumnDatas()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tables/base.go:381 +0x2b0
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*block).GetColumnDataByIds()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tables/blk.go:88 +0x3a4
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnBlock).GetColumnDataByIds()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/txn/txnimpl/block.go:280 +0x44c
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables/jobs.(*mergeBlocksTask).Execute()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tables/jobs/mergeblocks.go:231 +0x1653
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/ops.(*Op).OnExec()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/ops/ops.go:91 +0x1a2
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables/jobs.(*mergeBlocksTask).OnExec()
      <autogenerated>:1 +0x94
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*ScheduledTxnTask).Execute()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/db/task.go:64 +0x573
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/ops.(*Op).OnExec()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/ops/ops.go:91 +0x1a2
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*ScheduledTxnTask).OnExec()
      <autogenerated>:1 +0x94
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/worker.(*OpWorker).onOp()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/worker/worker.go:210 +0x95
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/worker.(*OpWorker).onOp-fm()
      <autogenerated>:1 +0x66
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*poolHandler).doHandle.func1.1()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/poolhandler.go:60 +0x8e
  github.com/panjf2000/ants/v2.(*goWorker).run.func1()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/worker.go:67 +0x1c4

Goroutine 216076 (running) created at:
  github.com/panjf2000/ants/v2.(*goWorker).run()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/worker.go:48 +0xe4
  github.com/panjf2000/ants/v2.(*Pool).retrieveWorker.func1()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:337 +0xef
  github.com/panjf2000/ants/v2.(*Pool).retrieveWorker()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:348 +0x2a6
  github.com/panjf2000/ants/v2.(*Pool).Submit()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:222 +0x9e
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*parallelJobScheduler).Schedule()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/job.go:98 +0xf6
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.(*IoPipeline).schedulerPrefetch()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/blockio/pipeline.go:430 +0x92
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.(*IoPipeline).onPrefetch()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/blockio/pipeline.go:473 +0x869
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/blockio.(*IoPipeline).onPrefetch-fm()
      <autogenerated>:1 +0x84
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/logstore/sm/safeq.go:89 +0x8ec

Goroutine 216072 (running) created at:
  github.com/panjf2000/ants/v2.(*goWorker).run()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/worker.go:48 +0xe4
  github.com/panjf2000/ants/v2.(*Pool).retrieveWorker.func1()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:337 +0xef
  github.com/panjf2000/ants/v2.(*Pool).retrieveWorker()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:348 +0x2a6
  github.com/panjf2000/ants/v2.(*Pool).Submit()
      /home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.4/pool.go:222 +0x9e
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*poolHandler).doHandle()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/poolhandler.go:65 +0x215
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks.(*poolHandler).doHandle-fm()
      <autogenerated>:1 +0x66
  github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/worker.(*OpWorker).Start.func1()
      /home/runner/_work/matrixone/matrixone/head/pkg/vm/engine/tae/tasks/worker/worker.go:142 +0x293
==================
```

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/7191461449

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/7176717477/job/19542506407

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/7177008172/job/19543013201

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/7177957596/job/19545580594

**Comment by sukki37** [comment]
https://github.com/matrixorigin/matrixone/actions/runs/7178905636/job/19548385383

**Comment by nnsgmsone** [comment]
fixed by #13458
